source/lac/petsc_matrix_base.cc

```diff
@@ -89,14 +89,23 @@ namespace PETScWrappers
     AssertThrow(ierr == 0, ExcPETScError(ierr));
   }
 
+  void
+  MatrixBase::assign_petsc_matrix(Mat A)
+  {
+    AssertThrow(last_action == ::dealii::VectorOperation::unknown,
+                ExcMessage("Cannot assign a new Mat"));
+    PetscErrorCode ierr =
+      PetscObjectReference(reinterpret_cast<PetscObject>(A));
+    AssertThrow(ierr == 0, ExcPETScError(ierr));
```

Comment:
```suggestion
    const PetscErrorCode ierr =
      PetscObjectReference(reinterpret_cast<PetscObject>(A));
      (void)ierr;
    AssertThrow(ierr == 0, ExcPETScError(ierr));
```

Reply:
Why this change? should the `(void)` casting be applied basically everywhere in the code and not just here?

Reply:
Oh, I misread the code. It uses `AssertThrow` instead of `Assert`.

`Assert` is optimized out in release mode, leaving the variable `ierr` assigned to but never read, and as a consequence compilers will warn about that. Saying `(void)ierr;` tells the compiler "We know, treat the variable as used." But `AssertThrow` stays in place in both debug and release mode, and so the variable is used in both cases.

In other words, forget about the suggestion.